### PR TITLE
[MRG] Fix __repr__ for Categorical dimensions with many options

### LINKED ISSUE #539: printing minimization result with categorical variable with more than 7 categories fails
I tried this (the stack trace below is with the current head 39528092868fcb24c598ff2dcd9f90b1b55a1d81 but I originally saw the problem with skopt 0.4):

```
import numpy as np
from skopt import dummy_minimize
from skopt.space import Categorical

def f(x):
    print "evaluating at",x
    return np.random.randn()

res = dummy_minimize(f,
                    [
                        Categorical([1,2,3,4,5,6,7,8])
                    ])

print res
```
which fails on the print statement with the following stack trace:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../anaconda/envs/test/lib/python2.7/site-packages/scipy/optimize/optimize.py", line 125, in __repr__
    for k, v in sorted(self.items())])
  File ".../anaconda/envs/test/lib/python2.7/site-packages/skopt/space/space.py", line 529, in __repr__
    return "Space([{}])".format(',\n       '.join(map(str, dims)))
  File ".../anaconda/envs/test/lib/python2.7/site-packages/skopt/space/space.py", line 434, in __repr__
    cats = self.categories[:3] + [_Ellipsis()] + self.categories[-3:]
TypeError: can only concatenate tuple (not "list") to tuple
```

when I remove the last element from the list (resulting in a list with just seven elements) given to `Categorical()`, the above code works without problems. 

Is this expected ?

### PR DESCRIPTION
**Pull request by betatim**
Fixes #539

### CODE REVIEW

**Comment by iaroslav-ai** [comment]
LGTM!

**Comment by betatim** [comment]
Ha! @pep8speaks I like you. Any idea how to make the line shorter?

**Comment by iaroslav-ai** [comment]
Heh :) Can @pep8speaks also suggest edits to code? :D

File: skopt/tests/test_space.py [hunk 1]
```diff
@@ -166,6 +166,15 @@ def test_categorical_transform_binary():
     assert_array_equal(ent_inverse, categories)
 
 
+@pytest.mark.fast_test
+def test_categorical_repr():
+    small_cat = Categorical([1, 2, 3, 4, 5])
```
Comment (iaroslav-ai) [inline skopt/tests/test_space.py]: Maybe
```python
toprint =  "Categorical(categories=(1, 2, 3, 4, 5), prior=None)"
assert small_cat.__repr__() == toprint
```
to make pep happy :)

Comment (betatim) [reply skopt/tests/test_space.py]: :-/

I went for
```python
assert (small_cat.__repr__() ==
            "Categorical(categories=(1, 2, 3, 4, 5), prior=None)")
```

**Comment by betatim** [comment]
[Yes it can](https://github.com/OrkoHunter/pep8speaks#features) suggest changes :)

**Comment by iaroslav-ai** [comment]
That is just awesome. Hope I did not confuse it with my comments.

File: skopt/tests/test_space.py (hunk 1 above)
Comment (iaroslav-ai) [reply skopt/tests/test_space.py]: Fine with me! :)

**Comment by iaroslav-ai** [comment]
Tests for the new change are passing on my machine. Any other changes? If not I merge this right now.

**Pull request merged**